File: src/SimpleStocker.Web/src/app/models/sale.model.ts
import { BaseModel } from './basemodel.model';
import { EPaymentMethod } from './enums/ePaymentMethod';
import { ESaleStatus } from './enums/eSaleStatus';
import { SaleItem } from './saleItem.model';

export class Sale extends BaseModel {
  items: SaleItem[] = [];
  totalAmount: number = 0;
  customerId: number = 0;
  discount: number = 0;
  paymentMethod: EPaymentMethod = EPaymentMethod.Pix;
  status: ESaleStatus = ESaleStatus.Pending;

  constructor() {
    super();
    this.customerId = 0;
    this.discount = 0;
    this.paymentMethod = EPaymentMethod.Pix;
    this.status = ESaleStatus.Pending;
    this.items = [];
  }
}

File: src/SimpleStocker.SaleApi/Models/SaleModel.cs
using SimpleStocker.SaleApi.Models.Enums;

namespace SimpleStocker.SaleApi.Models
{
    public class SaleModel
    {
        public long Id { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime UpdatedDate { get; set; }
        public List<SaleItemModel> Items { get; set; }
        public long ClientId { get; set; }
        public decimal TotalAmount { get; set; }
        public decimal Discount { get; set; }
        public EPaymentMethod PaymentMethod { get; set; }
        public ESaleStatus Status { get; set; }


    }
}
